subsys/testsuite/ztest/src/ztest_posix.c

```diff
@@ -0,0 +1,180 @@
+ *
+ * @param file Filename to check
+ * @returns Shortened filename, or @file if it could not be shortened
+ */
+const char *ztest_relative_filename(const char *file)
+{
+	const char *cwd;
+	char buf[200];
+
+	cwd = getcwd(buf, sizeof(buf));
+	if (cwd && strlen(file) > strlen(cwd) &&
+	    !strncmp(file, cwd, strlen(cwd)))
+		return file + strlen(cwd) + 1; /* move past the trailing '/' */
+	return file;
+}
+
+/**
+ * @brief Helper function to set list_tests
+ *
+ * @param value - Sets list_tests to value
+ */
+void ztest_set_list_test(bool value)
+{
+	list_tests = value;
+}
+
+/**
+ * @brief Helper function to get command line argument for listing tests
+ *
+ * @return true
+ * @return false
+ */
+bool z_ztest_get_list_test(void)
+{
+	return list_tests;
+}
+
+/**
+ * @brief Helper function to get command line test arguments
+ *
+ * @return const char*
+ */
+const char *ztest_get_test_args(void)
+{
+	return test_args;
+}
+
+/**
+ * @brief Lists registered unit tests in this binary
+ *
+ * @return int Number of tests in binary
+ */
+int z_ztest_list_tests(void)
+{
+	struct ztest_suite_node *ptr;
+	struct ztest_unit_test *test = NULL;
+	int test_count = 0;
+
+	for (ptr = _ztest_suite_node_list_start; ptr < _ztest_suite_node_list_end; ++ptr) {
+		while ((test = z_ztest_get_next_test(ptr->name, test)) != NULL) {
+			TC_PRINT("    %s::%s\n", test->test_suite_name, test->name);
+			test_count++;
+		}
+	}
+
+	/* List tests only once */
+	ztest_set_list_test(false);
+	return test_count;
+}
+
+/**
+ * @brief Determines if the test case should run based on test cases listed
+ *	  in the command line argument.
+ *
+ * Overrides implementation in ztest_new.c
+ *
+ * @param suite - name of test suite
+ * @param test  - name of unit test
+ * @return true
+ * @return false
+ */
+bool z_ztest_should_test_run(const char *suite, const char *test)
+{
+	bool run_test = false;
+	char *suite_test_pair = (char *)malloc(strlen(suite) + strlen(test) + 4);
+	char *suite_test_wild = (char *)malloc(strlen(suite) + 4);
+
+	sprintf(suite_test_pair, "%s::%s", suite, test);
+	sprintf(suite_test_wild, "%s::*", suite);
+
+	run_test = (test_args == NULL ||
+		    strstr(test_args, suite_test_pair) != NULL ||
+		    strstr(test_args, suite_test_wild) != NULL);
+
+	free(suite_test_pair);
+	free(suite_test_wild);
+	return run_test;
+}
+
+/**
+ * @brief Determines if the test suite should run based on test cases listed
+ *	  in the command line argument.
+ *
+ * Overrides implementation in ztest_new.c
+ *
+ * @param state The current state of the machine as it relates to the test
+ *		executable.
+ * @param suite Pointer to ztest_suite_node
+ * @return true
+ * @return false
+ */
+bool z_ztest_should_suite_run(const void *state, struct ztest_suite_node *suite)
+{
+	bool run_suite = true;
+	char *suite_test = (char *)malloc(strlen(suite->name) + 3);
+
+	sprintf(suite_test, "%s::", suite->name);
+
+	if (test_args != NULL && strstr(test_args, suite_test) == NULL) {
```

Comment:
similar comment about `strstr` usage here. need to use the commas as real delimiters.